ansible/roles/java-base/tasks/main.yml

```diff
@@ -109,36 +60,96 @@
         name: "{{ java_package_name }}"
         state: present
 
-- name: check if java is installed AIX
-  stat:
-    path: /home/iojs/jdk8u192-b12-jre/bin
-  register: java_exists
-  when: os|startswith("aix")
-
 - name: check if java is installed IBMi
   stat:
     path: /QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit/bin/java
   register: java_exists
   when: os|startswith("ibmi")
 
-- name: download java AIX
-  get_url:
-    url: https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u192-b12/OpenJDK8U-jre_ppc64_aix_hotspot_8u192b12.tar.gz
-    dest: /tmp/
-  tags: java
-  when: os|startswith("aix") and java_exists.stat.exists == False
-
-- name: unarchive java AIX
-  unarchive:
-    src: /tmp/OpenJDK8U-jre_ppc64_aix_hotspot_8u192b12.tar.gz
-    remote_src: yes
-    dest: /home/iojs
-  tags: java
-  when: os|startswith("aix") and java_exists.stat.exists == False
-
 - name: install webupd8 oracle java 8 extras
   when: java.rc > 0 and os == "ubuntu1404" and arch != "ppc64"
   package: name="{{item}}" state=present
   with_items:
     - ca-certificates
     - oracle-java8-set-default
+
+# Uses the AdoptOpenJDK API https://github.com/adoptopenjdk/openjdk-api-v3
+# to discover the most recent release for the given adoptopenjdk_version and
+# platform.
+- name: fetch adoptopenjdk metadata
+  register: adoptopenjdk_metadata
+  uri:
+    body_format: json
+    return_content: yes
+    status_code: 200
+    url: "https://api.adoptopenjdk.net/v3/assets/feature_releases/{{ adoptopenjdk_version }}/ga?architecture={{ adoptopenjdk_arch }}&image_type=jre&jvm_impl=openj9&os={{ adoptopenjdk_os }}&project=jdk&heap_size=normal&page_size=1&sort_method=DEFAULT&sort_order=DESC&vendor=adoptopenjdk"
+  when: use_adoptopenjdk == True
```

Comment:
I'm not sure we want to update to the latest every time. That can introduce change/churn as well as the possibility that some machines have different versions making it harder to debug issues.

I wonder if setting the version we install in the script, but then have it output a warning when there is a never version?

Reply:
Having said that, if we update to the latest version of Java on other platforms automatically then I'm ok with being consistent with that.  My experience though is that often when you install through yum, etc you get the version that was latest when the OS was cut and you won't necessarily get a very up to date versions.

Reply:
Also, in case there's confusion, this will be the latest Java 8 (it won't jump to, e.g. Java 11 without the variables being manually updated).

Reply:
@richardlau thanks for the clarification, if its the latest 8, then thats probably ok and consistent with what the package-upgrade does.